Services/UserService/UserService.cs

```diff
@@ -0,0 +1,124 @@
+﻿using AutoMapper;
+using webapi.Dtos.Users;
+using webapi.Models;
+
+namespace webapi.Services.UserService
+{
+    /// <summary>
+    /// Implementation of the user service.
+    /// </summary>
+    public class UserService : IUserService
+    {
+        // In-memory storage for users
+        private static List<Users> users = new List<Users>();
+
+        private readonly IMapper _mapper;
+
+        public UserService(IMapper mapper)
+        {
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// Retrieves all users.
+        /// </summary>
+        /// <returns>A list of all users.</returns>
+        public async Task<ServiceResponse<List<GetUserDto>>> GetAllUsers()
+        {
+            ServiceResponse<List<GetUserDto>> serviceResponse = new ServiceResponse<List<GetUserDto>>();
+            serviceResponse.Items = (users.Select(c => _mapper.Map<GetUserDto>(c))).ToList();
+            return serviceResponse;
+        }
+
+        /// <summary>
+        /// Retrieves a user by ID.
+        /// </summary>
+        /// <param name="id">The ID of the user to retrieve.</param>
+        /// <returns>The user with the specified ID, or null if not found.</returns>
+        public async Task<ServiceResponse<GetUserDto>> GetUserById(int id)
+        {
+            ServiceResponse<GetUserDto> serviceResponse = new ServiceResponse<GetUserDto>();
+            serviceResponse.Items = _mapper.Map<GetUserDto>(users.FirstOrDefault(c => c.Id == id));
+            return serviceResponse;
+        }
+
+        /// <summary>
+        /// Adds a new user.
+        /// </summary>
+        /// <param name="user">The user to add.</param>
+        /// <returns>A list of all users including the newly added user.</returns>
+        public async Task<ServiceResponse<List<GetUserDto>>> AddUser(AddUserDto newUser)
+        {
+            ServiceResponse<List<GetUserDto>> serviceResponse = new ServiceResponse<List<GetUserDto>>();
+
+            Users user = _mapper.Map<Users>(newUser);
+
+            // Assign a unique ID to the user
+            // user.Id = users.Max(u => u.Id) + 1;
+            user.Id = users.Count + 1;
```

Comment:
this condition is wrong in case in memory you have 5 records the last record is having id is 5 and we delete one record the count will be 4 and this condition give 5 in next record and 5 is not a unique id.

Reply:
Done

Reply:
check comments again